lib/model/folder_sendrecv.go

```diff
@@ -1200,6 +1200,13 @@ func (f *sendReceiveFolder) copierRoutine(in <-chan copyBlocksState, pullChan ch
 	}()
 
 	for state := range in {
+		select {
+		case <-f.ctx.Done():
+			state.fail(errors.Wrap(f.ctx.Err(), "folder stopped"))
+			out <- state.sharedPullerState
+		default:
+		}
```

Comment:
Again, the only lengthy procedure seems weak hashing, which already takes a context? Do you have evidence this helps?